### PR TITLE
Remove username block on custom tenants 

### PR DESCRIPTION
**Pull request by scrawfor99**
### Description
This PR resolves: https://github.com/opensearch-project/security-dashboards-plugin/issues/1244

Since its creation the tenant-switch-panel has enforced that 

```  
const customTenantOptions = tenants
    .filter((tenant) => {
      return tenant !== GLOBAL_TENANT_KEY_NAME && tenant !== username;
    })

...
```

This has been the same case the creation of the tenancy feature https://github.com/opensearch-project/security-dashboards-plugin/pull/374 and serves an unclear purpose. In order to resolve this issue, I am removing the `!== username` part of the evaluation. I have also contacted the original author to see if there was any reason for the original enforcement.

Existing tests should serve to validate this change. 

### Testing
[Please provide details of testing done: unit testing, integration testing and manual testing]
Existing tests should validate the tenancy behavior 

### Check List
- [ ] ~New functionality includes testing~
- [ ] ~New functionality has been documented~
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by scrawfor99 — COMMENTED** [review]
DO NOT MERGE (but please review :))

File: public/apps/account/tenant-switch-panel.tsx [hunk 1]
```diff
@@ -121,7 +121,7 @@ export function TenantSwitchPanel(props: TenantSwitchPanelProps) {
   };
   const customTenantOptions = tenants
     .filter((tenant) => {
-      return tenant !== GLOBAL_TENANT_KEY_NAME && tenant !== username;
+      return tenant !== GLOBAL_TENANT_KEY_NAME;
```
Comment (RyanL1997) [inline public/apps/account/tenant-switch-panel.tsx:124]: I think the original purpose of this username check is related to this:
https://github.com/opensearch-project/security-dashboards-plugin/blob/9abc57d415e8eedfc96d1bb399cfe118500a4d21/public/apps/configuration/utils/tenant-utils.tsx#L159. The user's private tenant is excluded from the custom tenant options for similar reasons as the global tenant check. The private tenant is handled by a dedicated radio button (PRIVATE_TENANT_RADIO_ID). Including it in the custom tenant options would be redundant.

However, I do see the issue here, https://github.com/opensearch-project/security-dashboards-plugin/issues/1244. I think we can think about how to handle the case when user wants to have a custom tenant with the same name as the `username`. Because the above link shows that the `username` in tenant can also be reference to user's private tenant.

**Comment by cwperks** [comment]
@scrawfor99 From the issue description, there is a naming convention on the `.kibana` indices.

i.e.
```
.kibana_<hash>_<tenant_name> .kibana_<hash>_<username>
```

Do you know what the value of `hash` is for these?

File: public/apps/account/tenant-switch-panel.tsx (hunk 1 above)
Comment (derek-ho) [reply public/apps/account/tenant-switch-panel.tsx:124]: +1

Comment (derek-ho) [reply public/apps/account/tenant-switch-panel.tsx:124]: Based on the issue description I think the issue may be that the index of the private tenant of the user with a particular name would "collide" with the index of the tenant with that same name. Let's get some more clarity around what kind of edge cases this could create before merging. Could a workaround be to suggest changing the name of the custom tenant since it is custom anyway?

**Comment by cliu123** [comment]
> @scrawfor99 From the issue description, there is a naming convention on the `.kibana` indices.
> 
> i.e.
> 
> ```
> .kibana_<hash>_<tenant_name> .kibana_<hash>_<username>
> ```
> 
> Do you know what the value of `hash` is for these?

IIRC, it is a hash of tenant name, but it'll be good to double check.

File: public/apps/account/tenant-switch-panel.tsx (hunk 1 above)
Comment (scrawfor99) [reply public/apps/account/tenant-switch-panel.tsx:124]: Hi @RyanL1997 and @derek-ho,

Yeah. I am not sure how things relate with the indices spun up for each user or tenant.

To Derek's point, you could certainly just have the tenant be named something else but this issue was specifically looking at making them same. I assume that was a requirement for the specific user in question.

Comment (scrawfor99) [reply public/apps/account/tenant-switch-panel.tsx:124]: If this is going to require a refactor of the tenant system--which is already being replaced--I suggest we just let the user know that they will have to name the tenant something else. Otherwise, I think a minor change like this would be fine.

I am not sure how to determine whether there are edge cases with the index names... I'd like to think that testing would catch any issues but we would be better off doing nothing than making the change if there is a concern that this would potentially leak data across the indices or crash a cluster.

Comment (scrawfor99) [reply public/apps/account/tenant-switch-panel.tsx:124]: After this change, I was able to log in as a user and a tenant of the same name. That being said, just because it worked once does not mean it will always succeed. If there is a chance that the incorrect index could be accessed then we would be better off continuing to restrict the tenant name from being the same as the user...

What makes me unclear on whether this is a legitimate concern is that there is nothing stopping a cluster from having a tenant with the same name as a user. Instead it is just that specific user accessing the custom tenant with its matching name. This would seem less like a concern about the user and tenant having the matching name but instead that the private tenant default has an index with the same name as the user.

Comment (RyanL1997) [reply public/apps/account/tenant-switch-panel.tsx:124]: > If this is going to require a refactor of the tenant system--which is already being replaced--I suggest we just let the user know that they will have to name the tenant something else. Otherwise, I think a minor change like this would be fine.

I agree. Or we can add a check for tenant creation to check if the given name is the same as the current `username`, However, the challenging part is to handle the edge case that if a user decides to name a custom tenant by using another user's `username`.. At least, for #1244, the workaround is to let the user rename their tenant into something irrelevant to any of the `username`s.

Comment (derek-ho) [reply public/apps/account/tenant-switch-panel.tsx:124]: I was able to create a use case that demonstrates the edge case that I think this is testing:
- As admin create a tenant "test-tenant"
- Switch to this tenant and create a dashboard
- Create a user called "test-tenant"
- Log in as this user
- Switch to "Private tenant"
- You can see the dashboard that was created by admin in the custom tenant as your own private tenant

Comment (scrawfor99) [reply public/apps/account/tenant-switch-panel.tsx:124]: Yes, I reproduced the same.

Comment (scrawfor99) [reply public/apps/account/tenant-switch-panel.tsx:124]: Screenshot 2024-06-10 at 4 58 23 PM

**Comment by scrawfor99** [comment]
Closing this PR since it is not feasible to resolve the index confusion between a tenant with a user's name and that user's private tenant.
